tests/commands/build/test_call.py

```diff
@@ -1217,3 +1217,123 @@ def test_test_app_unbuilt(build_command, first_app_unbuilt, second_app):
             {"update_state": "second", "build_state": "first", "test_mode": True},
         ),
     ]
+
+
+def test_build_app_single(build_command, first_app, second_app):
+    """If the --app flag is used, only the selected app is built."""
+    # Add two apps
+    build_command.apps = {
+        "first": first_app,
+        "second": second_app,
+    }
+
+    # Configure command line options
+    options, _ = build_command.parse_options(["--app", "first"])
```

Comment:
This test should be parameterised to confirm that both `--app` and `-a` work. (You could also duplicate the test - but given the test is otherwise identical, parameterization makes more sense.)